app/Helpers/database/user-activity.php

```diff
@@ -440,100 +443,165 @@ function GetMostPopularTitles(int $daysRange = 7, int $offset = 0, int $count =
     return $data;
 }
 
-function getUserGameActivity(string $user, int $gameID): array
+function getUserGameActivity(string $username, int $gameID): array
 {
-    sanitize_sql_inputs($user);
-
-    $query = "SELECT a.timestamp, a.lastupdate, a.data
-              FROM Activity a
-              WHERE a.User='$user' AND a.data=$gameID
-              AND a.activitytype=" . ActivityType::StartedPlaying;
-    $dbResult = s_mysql_query($query);
-    if ($dbResult === false) {
-        log_sql_fail();
+    $user = User::firstWhere('User', $username);
+    if (!$user) {
+        return [];
+    }
 
+    $game = Game::firstWhere('ID', $gameID);
+    if (!$game) {
         return [];
     }
 
+    $achievements = [];
+    $unofficialAchievements = [];
     $sessions = [];
-    while ($row = mysqli_fetch_assoc($dbResult)) {
-        $sessions[] = [
-            'StartTime' => strtotime($row['timestamp']),
-        ];
 
-        if ($row['lastupdate'] !== $row['timestamp']) {
-            $sessions[] = [
-                'StartTime' => strtotime($row['lastupdate']),
-            ];
+    $playerSessions = PlayerSession::where('user_id', '=', $user->ID)
+        ->where('game_id', '=', $gameID)
+        ->get();
+    foreach ($playerSessions as $playerSession) {
+        $session = [
+            'StartTime' => $playerSession->created_at->unix(),
+            'EndTime' => $playerSession->updated_at->unix(),
+            'IsGenerated' => $playerSession->created_at < Carbon::create(2023, 10, 14, 13, 16, 42),
```

Comment:
The timestamp is from the first `player_session` created in the prod database. 

This is leftover from when I tried to backfill `player_sessionss from the `Activity` data, which turned out to be unreasonably expensive (estimate to completely backfill the database was nearly a year due to querying the Activity table multiple times for every one of the roughly eight million player_games rows).